### ISSUE TITLE
Blocking and unblocking a user from a DM crashes the app

### ISSUE THREAD

**Issue by VolkerJunginger** [vector-im/element-x-ios#2140]
### Steps to reproduce

1. Where are you starting? What can you see?
Blocking/ unblocking  a user from the user-details in DM crashes the app.

3. What do you click?

- Open a DM (not a group-chat. This seems to work.)
- Go to user details
- Block / unblock user

#### What did you expect?
User gets blocked.

#### What happened instead?
App crashes.

### Application version

Nightly Version 1.4.2. (441)

### Will you send logs?

No

**Comment by pixlwave** [comment]
@VolkerJunginger Please can we have a rageshake on this one? Just like #1998 this isn't a simple 1-step reproduction, so there must be more at play here.

https://github.com/vector-im/element-x-ios/assets/6060466/c7e7929f-5958-430a-b129-4575519d6058

**Comment by kittykat** [comment]
Possibly a duplicate of https://github.com/vector-im/element-x-ios/issues/1998
